public/livesearch-autoparco-tipologie.php

```diff
@@ -0,0 +1,55 @@
+<?php
+    session_start();
+
+    if ($_SESSION['auth'] < 1 ) {
+        header("Location: index.php?s=1");
+        exit();
+    }
+
+    include '../db-connessione-include.php';
+    include 'maledetti-apici-centro-include.php'; //ATTIVA O DISATTIVA IL MAGIC QUOTE PER GLI APICI
+
+    function __autoload ($class_name) { //funzione predefinita che si occupa di caricare dinamicamente tutti gli oggetti esterni quando vengono richiamati
+        require_once "class/" . $class_name.".obj.inc";
+    }
+
+    $a = new Anagrafica();
+    $c = new Calendario();
+
+    $ogg = $_GET['q'];
+    $num = $_GET['num'];
+
+    $query = $connessione->query("SELECT * FROM aut_tipologie WHERE descrizione LIKE '%$ogg%' ORDER BY descrizione ASC LIMIT $num");
```

Comment:
Questo codice è vulnerabile ad attacchi di tipo SQL injection.

Non usare *mai* variabili direttamente in una query. Piuttosto, usa sempre i *prepared statements* di PDO, come segue:

```php
$query = $connessione->prepare("SELECT * FROM aut_tipologie WHERE descrizione LIKE :descrizione ORDER BY descrizione ASC LIMIT :num");
$query->bindValue(":descrizione", "%" . $ogg . "%");
$query->bindValue(":num", $num, PDO::PARAM_INT);
$query->execute();
```